### ISSUE TITLE
Why GIoU loss returns a (N,) tensor?

### ISSUE THREAD

**Issue by MSTK2014** [open-mmlab/mmdetection#7262]
GIoU loss returns a (N,) tensor, where N is the number of samples.  BBox regression needs a tensor (N, 4). Could you please tell me where the operation repeat(1, 4) take place?

**Comment by chhluo** [comment]
For GIoULoss, set `reduction = 'none'` will return a tensor of shape (N, ), which is unreduced.
```python
from mmdet.models import build_loss
import torch

loss_cfg = dict(
    type = "GIoULoss",
    reduction="mean"
)
loss = build_loss(loss_cfg)
pred = torch.tensor([[1, 2, 3, 4], [4, 5, 6, 7], [7,8,9,10]])
target = torch.tensor([[1.1, 2, 3, 4], [4,5.1,6,7], [3,4,5,6]])
out = loss(pred, target)

print(pred.shape, target.shape, out.shape, out)
# torch.Size([3, 4]) torch.Size([3, 4]) torch.Size([]) tensor(0.6259)
loss_cfg = dict(
    type = "GIoULoss",
    reduction='none'
)
loss = build_loss(loss_cfg)
out = loss(pred, target)
print(pred.shape, target.shape, out.shape, out)
# torch.Size([3, 4]) torch.Size([3, 4]) torch.Size([3]) tensor([0.0500, 0.0500, 1.7778])
```
For BBox regrssion, I didn't get it.
